.travis.yml

```diff
@@ -4,6 +4,11 @@ dist: trusty
 
 cache:
   apt: true
+  directories:
+    # Cache directory for older Composer versions.
```

Comment:
Travis always does a `composer self-update` - so why is the old directory needed?

Reply:
That question should probably have been asked when the PR adding this was pulled....

Reply:
I hadn't studied how Travis works much at that point. Now I have :-)

Reply:
Ok, well, my take on this is quite simple:
* Having the extra directory here makes no difference for Travis - if there is nothing in it, nothing will be cached anyway.
* Whether or not the `composer self-update` command should be part of the script is debatable. I'm on the "let's remove it" side as:
    1. Travis generally updates their images regularly enough that the version of Composer shipped with it will not be that much out of date anyway.
    2. If Travis - now or in the future - uses different versions of Composer on images targeting different PHP versions, all the better, as that means our set-up and composer validation would be tested on different versions. Self-update removes that testing.
* If the self-update command would be removed from the script, having both directories in the cache list is the better setup.

Reply:
> Whether or not the composer self-update command should be part of the script is debatable. I'm on the "let's remove it" side as:

The `composer self-update` I was referring to is [part of the environment setup](https://travis-ci.org/WordPress-Coding-Standards/WordPress-Coding-Standards/jobs/421374075#L439), well before our `before_install` scripts start kicking in. We can't remove their `composer self-update`, which is why the old extra path, I believe, is completely redundant.

Reply:
Ah, ok, that might actually be a semi-recent thing as I distinctly remember that Travis "in the olden days" (which may only be a couple of months ago) didn't do that and you had to add the command to your own script if you wanted to be sure to use the current version of Composer.